src/vipersci/vis/pds/create_raw.py

```diff
@@ -0,0 +1,403 @@
+                "hazlight_aft_port_on": False,
+                "hazlight_aft_starboard_on": False,
+                "hazlight_center_port_on": False,
+                "hazlight_center_starboard_on": False,
+                "hazlight_fore_port_on": False,
+                "hazlight_fore_starboard_on": False,
+                "navlight_left_on": False,
+                "navlight_right_on": False,
+                "mission_phase": "TEST",
+                "purpose": "Navigation",
+                "onboard_compression_ratio": 64,
+                "onboard_compression_type": parameter.name[-4:].upper()
+            }
+            d.update(parameter.eng_value['imageHeader'])
+            d["yamcs_name"] = parameter.name
+            d["yamcs_generation_time"] = parameter.generation_time
+
+            with io.BytesIO(parameter.eng_value['imageData']) as f:
+                im = imread(f)
+
+            self.__call__(d, im)
+
+
+def create(
+    metadata: dict,
+    image: Union[npt.NDArray[np.uint16], Path] = None,
+    outdir: Path = Path.cwd(),
+    session: Session = None,
+    template_path: Path = None
+):
+    """
+    Creates a PDS4 XML label file in *outdir* based on the provided
+    meta-data.  Returns None.
+
+    If *image* is a numpy array, that array will be considered the
+    Array_2D_Image and will be written to a TIFF file with the same
+    naming scheme as the XML file in *outdir*.  If *image* is a
+    file path to a TIFF file, the TIFF file at that path will be
+    evaluated and its meta-dataadded to the XML label, as if that
+    file was the File_Area_Observational for the XML label.  If
+    *image* is None (the default), no TIFF file will be written,
+    and certain elements of the XML label (pertaining to the
+    File_Area_Observational) will be empty.
+
+    If a path is provided to *outdir* the XML and optional TIFF
+    file will be written there. Defaults to the current working
+    directory.
+
+    If *session* is given, information for the raw product will be
+    written to the raw_products table.  If not, no database activity
+    will occur.
+
+    If *template_path* is passed to the write_xml() function.
+    """
+    rp = make_raw_product(metadata, image, outdir)
+
+    write_xml(rp.label_dict(), outdir, template_path)
+
+    if session is not None:
+        with session.begin() as s:
+            s.add(rp)
+
+    return
+
+
+def make_raw_product(
+    metadata: dict,
+    image: Union[npt.NDArray[np.uint16], Path] = None,
+    outdir: Path = Path.cwd(),
+) -> RawProduct:
+    """
+    Returns a Raw_Product created from the provided meta-data, and
+    if *image* is a numpy array, it will also use write_tiff() to
+    create a TIFF data product in *outdir* (defaults to current
+    working directory).
+    """
+    rp = RawProduct(**metadata)
+
+    if image is not None:
+        if isinstance(image, Path):
+            # Should we bother to test filename patterns against the pid?
```

Comment:
Yeah, so this `--tiff` functionality is a really just for lazy testing.  If you don't want to have create_raw actually take a source image (of almost any kind) and recast its pixels into a TIFF product, you can give it an existing `--tiff` file.  If we got fussy about this TIFF's name, then it would have to match the Product ID pattern of the product you're trying to make.  I am going to remove the comment, because I don't think we should bother for this usage.

We can always impose a test if we have problems.